lib/rules/sort-keys.js

```diff
@@ -156,13 +177,24 @@ module.exports = {
                 }
 
                 const prevName = stack.prevName;
+                const prevEndLine = stack.prevEndLine;
                 const numKeys = stack.numKeys;
                 const thisName = getPropertyName(node);
+                const thisStartLine = node.loc.start.line;
+                const isLineSeparatedGroups = prevEndLine && checkLineSeparatedGroups(thisStartLine, prevEndLine);
+
+                if (thisStartLine !== null) {
+                    stack.prevEndLine = thisStartLine;
```

Comment:
It seems to treat the multiline property as a group separator because it compares the `start line of the previous property` and `the start line of the current property`. So the below case has no lint error.

```js
{
   code: `
     var obj = {
        b: 1, // 1st Group
        c () { // 1st Group 

        },
        a: 3 // 2nd Group ??
      }
     `,
    options: ["asc", { allowLineSeparatedGroups: true }],
    parserOptions: { ecmaVersion: 6 }
},
```

I think we should compare the end line of the previous property with the start line of the current property as what "sort-import" does

* [demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgc29ydC1pbXBvcnRzOiBbXCJlcnJvclwiLCB7IFwiYWxsb3dTZXBhcmF0ZWRHcm91cHNcIjogdHJ1ZSB9XSovXG5cbmltcG9ydCBiIGZyb20gJ2Zvby5qcyc7XG5pbXBvcnQgZCBcbmZyb20gJ2Jhci5qcyc7XG5pbXBvcnQgYSBmcm9tICdiYXouanMnOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTEsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)
```js
/*eslint sort-imports: ["error", { "allowSeparatedGroups": true }]*/

/*eslint sort-imports: ["error", { "allowSeparatedGroups": true }]*/

import b from 'foo.js';
import d 
   from 'bar.js';
import a from 'baz.js'; // Lint error, cause it's not separated from previous
```

Reply:
Sorry !!!! 
I made a mistake while clean the code.

It was `stack.prevEndLine = node.loc.end.line;` before PR.

Thank you !

Reply:
> Sorry !!!!
I made a mistake while clean the code.

There's no need to be sorry :) 
Can we add a test case for ensuring it? (maybe in invalid cases?)

```js
{
   code: `
     var obj = {
        b: 1, 
        c () { 

        },
        a: 3
      }
     `,
    options: ["asc", { allowLineSeparatedGroups: true }],
    parserOptions: { ecmaVersion: 6 }
},
```